### DISCUSSION TITLE
Cast/Crew Design Idea

### DISCUSSION THREAD

**Discussion by Perseusdehond** [jellyfin/jellyfin-vue#516] [UI & UX]
Hello there again, I made a redesign of the Cast/Crew item page because as of right now there isn’t really any. I tried again to make this design in-line of what is already there, but without going to another page like is the case in Jellyfin web. For this design I used Pirates of the Caribbean as an example. I especially want to thank Tiwenty for helping me improve the design to what it is now.

When you are on an item page you could click on an actor/writer/director and it would show his/her information in the section that the movie/serie was in. This way I think you don’t have too much empty space, because the Cast/Crew bar is still there. The actor could be displayed in a rectangle instead of a circle (this is already the case in JF web). With an blue (color can be different) bar it shows what actor you just clicked on and shows him. Then it says which role he plays in set movie together with the birth date and place. I find it fun to know how hold someone is without having to calculate that, so that would be a nice addition Imo. Then it also just gives a short bio where the border is set until the end of the picture card on bigger screens I would say. It also shows the other movies the actor/director was a part of. Displaying it this way would allow users to quickly switch between Cast/Crew members without unnecessary clicks. By clicking on the selected actor again, the information about him would also go away and it would show the movie item screen again. This is seen in the top section of my design.

Then in the bottom part/half of the design I think it would be handy for a faster look at all the Cast/Crew members that you could click on Cast/Crew and then you would see all the descriptions of the members involved from Cast or Crew. I wouldn’t show the recommended or related content, because this would feel too crowded or too much of a scroll to get through and you can get it already by clicking on the member or clicking on full info page. After some rethinking, we decided to go with minimalistic information about set Cast/Crew member and have multiple columns depending on the size of the screen or how many members there are to show.

![Full Cast Crew Item Page](https://user-images.githubusercontent.com/42120256/104006312-9cadf800-51a6-11eb-90ef-4f8b2a5596fe.png)

On a logic side, this would open a new 'page'. You could get back to the movie by clicking on the selected Cast/Crew member or Cast/Crew button again. You could also just click on the back arrow, so two easy ways to get in and out of these screens.
Think this explanation is enough, hope you like it ;)

**Comment by ThibaultNocchi** [comment]
I'm just commenting that I still like your idea and will implement this sooner or later :)

**Comment by ferferga** [comment]
I agree with everything except the left scrollbar in cast and crew. I don't think we can't even do that and, even if we could it looks counter intuitive to go against the flow that already exists (scrollbars at the right side)

I wouldn't use any scrollbar for that though anyway.

**Reply by ThibaultNocchi** [reply]
You mean the blue bar at the left of Johnny Depp? I didn't think of it as a scroll bar, more as an active indicator. But yeah this bar here wouldn't be easy ^^

**Reply by Perseusdehond** [reply]
Yeah it isn't a scrollbar, but an indication of what actors you have selected/are shown. I put it at the left side because all the names start at the same point, but they don't end at the same point so that would look a bit worse imo. I also made it to look a bit like how the blue line appears underneath a selected section on for example the artist page. You could use another way to show which actor is selected in case a single actor is selected like how it shows that cast is selected (grey it out). But I don't really mind if that bar goes away ;)

**Reply by ThibaultNocchi** [reply]
Oh, disabling the current actor could be done, as we'll probably use a vuetify list, items can easily be disabled, so we'll see :)
